lib/Doctrine/ruleset.xml

```diff
@@ -419,6 +419,14 @@
             </property>
         </properties>
     </rule>
+    <!-- Define unions style -->
+    <rule ref="SlevomatCodingStandard.TypeHints.UnionTypeHintFormat">
+        <properties>
+            <property name="withSpaces" value="no" />
+            <property name="shortNullable" value="no" />
```

Comment:
What's the rationale behind this one? I think I would have gone with `yes`.

Reply:
Consistency here. `?|type` got slightly redundant IMO.

**All unions should be written the same**

- `int|null`
- `int|string`

 I did not found any reason for having an exception for nulls. Why write `?int` when we can write `int|null` the same way as we write `int|string`.

**Union extension**

User can append new type:
```diff
- int|null
+ int|string|null
```

instead of having to rewrite it

```diff
- ?int
+ int|string|null
```

Reply:
The Union extension is not very convincing, I don't think that's something that happens often (having to extend a union type). Consistency is more convincing.

I have one question though: what will happen if somebody tries to use this on a PHP version that does not support union types?

Reply:
that's covered in tests. You can see it applies to phpdoc.

https://github.com/doctrine/coding-standard/pull/266/files#diff-5ae842674196e659607e3a022a3ffe34f75a77e7616a8052d47b79412cb84bef

https://github.com/doctrine/coding-standard/pull/266/files#diff-2332acfd1c5d720a157ae9c82bacd8afb47af0ce4e2ffdea0755b9dec288acb8

Reply:
In case nothing happens, I think this might be fine, but should probably go to 10.0.x given how much code it will change.

Reply:
Yup, it only sorts types (puts null (empty value) at the end).

I did not notice there's 10.0.x branch already. Will retarget when I'm at pc.

Reply:
> I did not notice there's 10.0.x branch already

I just created it ;)